apps/browser/qml/pages/components/PopUpMenu.qml

```diff
@@ -30,74 +30,112 @@ SilicaControl {
 
+        active: false
         source: menuComponent
 
         onAnimate: {
             if (item) {
-                item.opacity = 0
-                menuFadeAnimation.target = item
-                menuFadeAnimation.from = 0
-                menuFadeAnimation.to = 1
-                menuFadeAnimation.restart()
+                item.percentageClosed = 1
+                menuAnimation.target = item
+                menuAnimation.from = item.percentageClosed
+                menuAnimation.to = 0
+                menuAnimation.restart()
             } else if (replacedItem) {
-                menuFadeAnimation.target = replacedItem
-                menuFadeAnimation.from = 1
-                menuFadeAnimation.to = 0
-                menuFadeAnimation.restart()
+                menuAnimation.target = replacedItem
+                menuAnimation.from = 0
+                menuAnimation.to = 1
+                menuAnimation.restart()
             }
         }
     }
 
-    FadeAnimation {
-        id: menuFadeAnimation
+    NumberAnimation {
+        id: menuAnimation
 
         running: false
+        duration: 2000
+        easing.type: Easing.InOutQuad
+        property: "percentageClosed"
     }
 
     Component {
         id: menuComponent
 
         Item {
+            id: menuItem
+
+            property real percentageClosed
+            readonly property real menuTop: Math.max(0, headerItem.y - menuFlickable.contentY)
+
             width: popUpMenu.width
             height: popUpMenu.height
 
             SilicaFlickable {
                 id: menuFlickable
 
-                width: popUpMenu.width
-                height: popUpMenu.height
+                x: popUpMenu.width - width - popUpMenu.margin
+                y: popUpMenu.height
+                        - height
+                        - popUpMenu.margin
+                        + (menuItem.percentageClosed * (height - menuItem.menuTop + popUpMenu.margin))
+
+                width: Math.max(
+                            Theme.paddingLarge * widthRatio,
+                            footerLoader.item ? footerLoader.item.implicitWidth : 0)
+                height: Math.min(
+                            popUpMenu.height - (2 * popUpMenu.margin),
+                            headerItem.height + contentLoader.height + footerLoader.height)
 
                 contentHeight: contentLoader.y + contentLoader.height + footerLoader.height
 
                 interactive: popUpMenu.active   // Don't handle mouse events during fade out.
 
+                Item {
+                    id: headerItem
+
+                    y: Math.max(0, footerLoader.y - height - (Screen.sizeCategory > Screen.Medium
+                                ? contentLoader.height
+                                : Math.min(contentLoader.height, Theme.paddingLarge * popUpMenu.heightRatio)))
+
+                    width: menuFlickable.width
+                    height: Theme.paddingLarge
+
+                    Rectangle {
```

Comment:
Let's raise this above actual content items. On landscape menu expands to the max position leaving only panning of the flickable. When panning menu downwards, you can see all menu items going above the handle

Reply:
Done if I've understood correctly.